extension/encoding/awslogsencodingextension/internal/unmarshaler/s3-access-log/unmarshaler.go

```diff
@@ -0,0 +1,244 @@
+			continue
+		}
+
+		if i == fieldIndexTime {
+			// This is the timestamp that follows a strict format
+			// "[DD/MM/YYYY:HH:mm:ss zone]". Since zone is after a
+			// space, we cut the string again to remove the zone.
+			// Zone is always UTC, so it will always be +0000.
+			_, remaining, _ = strings.Cut(remaining, " ")
+		}
+
+		if err = addField(i, value, resourceAttr, record); err != nil {
+			return err
+		}
+	}
+
+	if i != fieldIndexACLRequired+1 {
+		return errors.New("values in log line are less than the number of available fields")
+	}
+
+	rScope := scopeLogs.LogRecords().AppendEmpty()
+	record.MoveTo(rScope)
+
+	return nil
+}
+
+func addField(field int, value string, resourceAttr *resourceAttributes, record plog.LogRecord) error {
+	switch field {
+	case fieldIndexS3BucketOwner:
+		resourceAttr.bucketOwner = value
+	case fieldIndexS3BucketName:
+		resourceAttr.bucketName = value
+	case fieldIndexTime:
+		// The format in S3 access logs at this point is as "[DD/MM/YYYY:HH:mm:ss".
+		if value == "" {
+			return errors.New("unexpected: time value is empty")
+		}
+		t, err := time.Parse("02/Jan/2006:15:04:05", value[1:])
+		if err != nil {
+			return fmt.Errorf("failed to get timestamp of log: %w", err)
+		}
+		record.SetTimestamp(pcommon.NewTimestampFromTime(t))
+	case fieldIndexHTTPStatus,
+		fieldIndexBytesSent,
+		fieldIndexObjectSize,
+		fieldIndexTurnAroundTime,
+		fieldIndexTotalTime:
+		n, err := strconv.ParseInt(value, 10, 64)
+		if err != nil {
+			return fmt.Errorf("value for field %q in log line is not a number", field)
+		}
+		attrName := attributeNames[field]
+		record.Attributes().PutInt(attrName, n)
+	case fieldIndexACLRequired:
+		attrName := attributeNames[field]
+		// If the request required an ACL for authorization, the string is Yes.
+		// If no ACLs were required, the string is -.
+		switch value {
+		case "Yes":
+			record.Attributes().PutBool(attrName, true)
+		case unknownField:
+			record.Attributes().PutBool(attrName, false)
+		default:
+			return fmt.Errorf("unknown value %q for field %q", value, field)
+		}
+	case fieldIndexTLSVersion:
+		// The value is one of following: TLSv1.1, TLSv1.2, TLSv1.3.
+		// We get the version after "v".
+		_, remaining, _ := strings.Cut(value, "v")
+		if remaining == "" {
+			_, remaining, _ = strings.Cut(value, "V")
+			if remaining == "" {
+				return fmt.Errorf("missing TLS version: %q", value)
+			}
+		}
+		attrName := attributeNames[fieldIndexTLSVersion]
+		record.Attributes().PutStr(attrName, remaining)
+	case fieldIndexRequestURI:
+		method, remaining, _ := strings.Cut(value, " ")
+		if method == "" {
+			return fmt.Errorf("unexpected: request uri %q has no method", value)
+		}
+		record.Attributes().PutStr(semconv.AttributeHTTPRequestMethod, method)
+		path, remaining, _ := strings.Cut(remaining, " ")
+		if path == "" {
+			return fmt.Errorf("unexpected: request uri %q has no path", value)
+		}
+		record.Attributes().PutStr(semconv.AttributeURLPath, path)
```

Comment:
I notice that the example at https://docs.aws.amazon.com/AmazonS3/latest/userguide/LogFormat.html includes a query string (`?x-foo=bar`). That should not be put into `url.path`.

We should probably use https://pkg.go.dev/net/url#ParseRequestURI and extract the non-empty parts.

Reply:
I have updated the fields and the README.